ihe-interop/src/main/java/org/motechproject/ihe/interop/handler/helper/IHEActionHelper.java

```diff
@@ -70,5 +74,20 @@ public void handleAction(Map<String, Object> parameters) throws IOException, Cla
         Template xmlTemplate = xmlTemplateEngine.createTemplate(new String(ArrayUtils.toPrimitive(templateData)));
         Writable writable = xmlTemplate.make(parameters);
         LOGGER.info("Template with name {}:\n{}", cdaTemplate.getTemplateName(), writable.toString());
+        sendTemplate(hl7Recipient.getRecipientUrl(), writable.toString());
+    }
+
+    private void sendTemplate(String url, String template) throws IOException{
```

Comment:
can you create this method in separate service ? Here should be only invocation of method from this new service